NSAttributedString+Markdown.m

```diff
@@ -541,6 +571,74 @@ static void updateAttributedString(NSMutableAttributedString *result, NSString *
+                                case 0:
+                                    bulletCharacter = bulletLevel1;
+                                    break;
+                                case 1:
+                                    bulletCharacter = bulletLevel2;
+                                    break;
+                                default:
+                                    bulletCharacter = bulletLevelGreaterThan2;
+                                    break;
+                            }
+                    
+                            UIFont *baseFont = [result attribute:NSFontAttributeName atIndex:(beginRange.location - mutationOffset) effectiveRange:NULL];
+                            UIFont *updatedBulletFont = [UIFont fontWithName:bulletFontName size:baseFont.pointSize];
+                            NSRange mutatedBeginRange = NSMakeRange(beginRange.location - mutationOffset - spacesBeforeBullet, beginRange.length + spacesBeforeBullet);
+                            [result replaceCharactersInRange:mutatedBeginRange withString:[NSString stringWithFormat:@"%@%@", bulletCharacter, @"\t"]];
+                            mutationOffset += spacesBeforeBullet;
+                            
+                            /*
+                             NSMutableAttributedString Bug? - apply bullet font adjustment AFTER replacing text or NSAttributedString will apply font
+                             change to beginning of list item text for mutationOffset # of characters
+                             */
+                            [result addAttribute:NSFontAttributeName value:updatedBulletFont range:NSMakeRange(mutatedBeginRange.location, 1)];
+                            
+                            NSRange listItemRangeWithBullet = NSMakeRange(mutatedBeginRange.location, endRange.location - mutatedBeginRange.location + endRange.length - mutationOffset - 1);
+                            NSMutableParagraphStyle *paragraphStyle = [[result attribute:NSParagraphStyleAttributeName atIndex:listItemRangeWithBullet.location effectiveRange:NULL] mutableCopy];
+                            CGFloat bulletIndent = bulletIndentWidth * bulletLevel;
+                            CGFloat bulletedTextIndent = bulletIndentWidth * (bulletLevel + 1);
+                            NSTextTab *firstTabStop = [[NSTextTab alloc] initWithTextAlignment:NSTextAlignmentLeft location:bulletedTextIndent options:@{}];
+                            paragraphStyle.tabStops = @[firstTabStop];
+                            [paragraphStyle setFirstLineHeadIndent:bulletIndent];
+                            [paragraphStyle setHeadIndent:bulletedTextIndent];
+                            [result addAttribute:NSParagraphStyleAttributeName value:paragraphStyle range:listItemRangeWithBullet];
+                            replaceMarkers = NO;
+#endif
+                            break;
+                        }
+                        case MarkdownSpanUnderline: {
```

Comment:
Would be easier to read/review if the `MarkdownSpanUnderline` case were after the MarkdownSpanEmphasisDouble case, since the implementation is identical

Reply:
Good idea!